salt/returners/redis_return.py

```diff
@@ -171,7 +171,12 @@ def get_jids():
     Return a list of all job ids
     '''
     serv = _get_serv(ret=None)
-    return list(serv.smembers('jids'))
+    ret = {}
+    for s in serv.mget(serv.smembers('jids')):
+        load = json.loads(s)
```

Comment:
I tested this out on a 2015.8.1 master running python 2.7.6 (ubuntu 14.04, distro python).  After running > 1 highstate across a few nodes:
```
ubuntu@peter.norton-saltmaster-vagrant:~$ sudo salt-run jobs.list_jobs
Exception occurred in runner jobs.list_jobs: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/client/mixins.py", line 337, in low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/runners/jobs.py", line 266, in list_jobs
    ret = mminion.returners['{0}.get_jids'.format(returner)]()
  File "/usr/lib/python2.7/dist-packages/salt/returners/redis_fix.py", line 176, in get_jids
    load = json.loads(s)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer
```

Adding a print statement provides this:

```
{"comment": "Version 2015.8.1+ds-1 of package 'salt-cloud' is already installed", "name": "salt-cloud", "start_time": "19:52:01.106230", "result": true, "duration": 1.049, "__run_num__": 77, "changes": {}}, "pkg_|-python-b
oto_|-python-boto_|-installed": {"comment": "Package python-boto is already installed", "name": "python-boto", "start_time": "19:52:01.185179", "result": true, "duration": 0.814, "__run_num__": 92, "changes": {}}, "file_|-
/etc/papertrail.crt_|-/etc/papertrail.crt_|-managed": {"comment": "File /etc/papertrail.crt is in the correct state", "name": "/etc/papertrail.crt", "start_time": "19:52:00.120695", "result": true, "duration": 4.531, "__ru
n_num__": 20, "changes": {}}, "file_|-/etc/mailname_|-/etc/mailname_|-managed": {"comment": "File /etc/mailname is in the correct state", "name": "/etc/mailname", "start_time": "19:52:00.274337", "result": true, "duration"
: 1.689, "__run_num__": 38, "changes": {}}, "file_|-/srv/reactor/sync_grains.sls_|-/srv/reactor/sync_grains.sls_|-managed": {"comment": "File /srv/reactor/sync_grains.sls is in the correct state", "name": "/srv/reactor/syn
c_grains.sls", "start_time": "19:52:01.191920", "result": true, "duration": 4.193, "__run_num__": 98, "changes": {}}, "service_|-remote_syslog_service_|-remote_syslog_|-running": {"comment": "The service remote_syslog is a
lready running", "name": "remote_syslog", "start_time": "19:52:00.147475", "result": true, "duration": 26.119, "__run_num__": 25, "changes": {}}, "cmd_|-/sys/module/nf_conntrack/parameters/hashsize_|-echo 24576 > /sys/modu
le/nf_conntrack/parameters/hashsize_|-run": {"comment": "unless execution succeeded", "name": "echo 24576 > /sys/module/nf_conntrack/parameters/hashsize", "start_time": "19:52:00.539685", "skip_watch": true, "result": true
, "duration": 9.029, "__run_num__": 65, "changes": {}}, "sysctl_|-net.core.rmem_max_|-net.core.rmem_max_|-present": {"comment": "Sysctl value net.core.rmem_max = 16777216 is already set", "name": "net.core.rmem_max", "star
t_time": "19:52:00.372155", "result": true, "duration": 16.663, "__run_num__": 54, "changes": {}}, "file_|-/etc/postfix/main.cf_|-/etc/postfix/main.cf_|-managed": {"comment": "File /etc/postfix/main.cf is in the correct st
ate", "name": "/etc/postfix/main.cf", "start_time": "19:52:00.269973", "result": true, "duration": 4.16, "__run_num__": 37, "changes": {}}}, "retcode": 0, "success": true, "tgt": "saltmaster", "cmd": "_return", "arg": [],
"fun": "state.highstate", "id": "saltmaster", "out": "highstate"}
{"tgt_type": "glob", "jid": "20151228195230433528", "cmd": "publish", "tgt": "*", "kwargs": {"delimiter": ":", "show_timeout": true, "show_jid": false}, "ret": "", "user": "sudo_ubuntu", "arg": [], "fun": "state.highstate"
}
{"tgt_type": "glob", "jid": "20151228195240495105", "tgt": "*", "cmd": "publish", "ret": "", "user": "sudo_ubuntu", "arg": ["20151228195230433528"], "fun": "saltutil.find_job"}
None
Exception occurred in runner jobs.list_jobs: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/client/mixins.py", line 337, in low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/runners/jobs.py", line 266, in list_jobs
    ret = mminion.returners['{0}.get_jids'.format(returner)]()
  File "/usr/lib/python2.7/dist-packages/salt/returners/redis_fix.py", line 177, in get_jids
    load = json.loads(s)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer
```

In testing I'm catching TypeError, and this succeeds after that.

How should returners run on the master log errors?  And is this None expected from redis?